cmake/BLTMacros.cmake

```diff
@@ -882,3 +882,66 @@ macro(blt_find_libraries)
     endforeach()
 
 endmacro(blt_find_libraries)
+
+
+##------------------------------------------------------------------------------
+## blt_combine_static_libraries( NAME <libname>
+##                               SOURCE_LIBS [lib1 ...] 
+##                               LIB_TYPE [STATIC,SHARED}
+##                               LINK_PREPEND []
+##                               LINK_POSTPEND []
+##                             )
+##
+## Adds a library target, called <libname>, to be built from the set of 
+## static libraries given in SOURCE_LIBS.
+## 
+## The LINK_PREPEND argument will be prepended to the library on the link line,
+## while the LINK_POSTPEND will be appended to the libray on the link line. These
+## values are typically "-Wl,--whole-archive", and "-Wl,--no-whole-archive" 
+## on linux systems
+##
+##------------------------------------------------------------------------------
+macro(blt_combine_static_libraries)
+
+    set(options )
+    set(singleValueArgs NAME LINK_PREPEND LINK_POSTPEND LIB_TYPE )
+    set(multiValueArgs SOURCE_LIBS )
+
+    cmake_parse_arguments(arg
+        "${options}" "${singleValueArgs}" "${multiValueArgs}" ${ARGN} )
+
+    # sanity checks
+    if( "${arg_NAME}" STREQUAL "" )
+        message(FATAL_ERROR "blt_combine_static_libraries() must be called with argument NAME <name>")
+    endif()
+
+    if(TARGET ${arg_NAME})
+        message(FATAL_ERROR "$A target with the name {arg_NAME} already exists!")
+    endif()
+
+    if( NOT arg_SOURCE_LIBS )
+        message(FATAL_ERROR "blt_combine_static_libraries(NAME ${arg_NAME} ...) called with no given source libraries")
+    endif()
+
+    set( libLinkLine "" )
+    foreach( lib ${arg_SOURCE_LIBS} )
+        list( APPEND libLinkLine ${arg_LINK_PREPEND} ${lib} ${arg_LINK_POSTPEND} )
+    endforeach()
+
+    if( ${arg_LIB_TYPE} STREQUAL "STATIC" )
+        set( _lib_type STATIC )
+    elseif( ${arg_LIB_TYPE} STREQUAL "SHARED" ) 
+        set( _lib_type SHARED )
+    else()
+        message(FATAL_ERROR "blt_combine_static_libraries(NAME ${arg_NAME} ...) LIB_TYPE must be SHARED OR STATIC")
+    endif()
+        
+    add_library ( ${arg_NAME} ${_lib_type} ${BLT_ROOT_DIR}/tests/internal/src/combine_static_library_test/dummy.cpp)
+    target_link_libraries( ${arg_NAME} PRIVATE ${libLinkLine})
+    blt_register_library( NAME ${arg_NAME}
```

Comment:
For some reason, this doesn't seem to be handling the transitive include dependencies correctly.

I configured cmake-3.9.2 using:
```bash
cmake  \ 
   -C ../host-configs/llnl-quartz-toss3-gcc@4.9.3.cmake  \
   -DCMAKE_EXPORT_COMPILE_COMMANDS=ON  \ 
   ../tests/internal/src/combine_static_library_test/
```
Here is the output when building ``FooStatic.x``:
```bash
[ 88%] Building CXX object CMakeFiles/FooStatic.x.dir/main.cpp.o
/usr/tce/packages/gcc/gcc-4.9.3/bin/g++    \
    -Wall -Wextra  -fPIE   -std=c++11 \
    -o CMakeFiles/FooStatic.x.dir/main.cpp.o \
    -c <_to_blt>/tests/internal/src/combine_static_library_test/main.cpp
<_to_blt>/tests/internal/src/combine_static_library_test/main.cpp:2:20: fatal error: foo1.hpp: No such file or directory
 #include "foo1.hpp"      
```
Note that there is no include flag to the path of ``"foo1.hpp"``

Reply:
sorry. I hadn't run it on unix. MacOSX filesystem isn't case sensitive....so it found the files that were upper case.

Reply:
Thanks for looking into it.  That makes sense.
After you fix this, could you please add this new test project to the travis plan?